packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx

```diff
@@ -195,26 +195,34 @@ export class GeneratedAnswerCommon {
     const {citations} = getGeneratedAnswerState() ?? {};
     const {logCitationHover} = getGeneratedAnswer() ?? {};
 
-    return citations?.map(
+    const uniqueUris = new Set<string>();
+
+    return citations?.flatMap(
```

Comment:
Why isn't this handled in Headless?

Reply:
@erocheleau Very good point! I honestly didnt think about it, as we are changing also a little bit the UI in atomic I did it here.
I could still change to headless but we still need to change the numbered citations in the UI for both!

Reply:
This branch should probably be based on #4250 now, I reckon, and the changes on Atomic to take into consideration those made in Headless.